File: exercise/week01/java/src/test/java/neural/Submarine.java
package neural;

import java.util.List;

public class Submarine {
    private int horizontal = 0;
    private int depth = 0;

    public void execute(String command) {
        String[] parts = command.split(" ");
        String action = parts[0];
        int value = Integer.parseInt(parts[1]);

        switch (action) {
            case "forward":
                horizontal += value;
                break;
            case "down":
                depth += value;
                break;
            case "up":
                depth -= value;
                break;
        }
    }

    public void executeAll(List<String> commands) {
        for (String command : commands) {
            if (command != null && !command.isBlank()) {
                execute(command);
            }
        }
    }

    public int getHorizontal() {
        return horizontal;
    }

    public int getDepth() {
        return depth;
    }

    public int finalPosition() {
        return horizontal * depth;
    }
}

File: exercise/week01/ts/Submarine.ts
export class Submarine {
    horizontal = 0;
    depth = 0;

    execute(command: string): void {
        const [action, valueStr] = command.split(' ');
        const value = parseInt(valueStr, 10);

        switch (action) {
            case 'forward':
                this.horizontal += value;
                break;
            case 'down':
                this.depth += value;
                break;
            case 'up':
                this.depth -= value;
                break;
        }
    }

    executeAll(commands: string[]): void {
        for (const command of commands) {
            if (command && command.trim() !== '') {
                this.execute(command);
            }
        }
    }

    finalPosition(): number {
        return this.horizontal * this.depth;
    }
}